src/clients/SpokePoolClient.ts

```diff
@@ -751,6 +778,22 @@ export class SpokePoolClient extends BaseAbstractClient {
           fill.relayExecutionInfo.updatedMessageHash = getMessageHash(event.args.relayExecutionInfo.updatedMessage);
         }
 
+        // Sanity check that this event is not a duplicate.
+        if (
+          this.fills[fill.originChainId] !== undefined &&
+          this.fills[fill.originChainId].some(
+            (f) => f.transactionHash === fill.transactionHash && f.logIndex === fill.logIndex
+          )
```

Comment:
Prettier probably complains about this formatting, but it should be possible to drop the first check.

```suggestion
        if (
          this.fills[fill.originChainId]?
            .some((f) => f.transactionHash === fill.transactionHash && f.logIndex === fill.logIndex)
```